### ISSUE TITLE
Filter on Volunteer Roster does not work due to custom report type

### ISSUE THREAD

**Issue by judisohn** [SalesforceFoundation/Volunteers-for-Salesforce#280]
[Thread 1](https://powerofus.force.com/0D58000003KE3ka) and [2](https://powerofus.force.com/0D58000003HU1bU) (same customer).

Replicated in test org. I believe the challenge is that we changed the custom report type to a WITH or WITHOUT join, and the filter is on the child object. See this [Help & Training document](https://help.salesforce.com/articleView?id=000004402&language=en_US&type=1), and [related blog post](http://www.simplysfdc.com/2015/02/salesforce-report-with-filter-on-child.html). Need to rethink how this report is generated... perhaps with cross filter as recommended?

**Comment by judisohn** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000005kjeNAAQ" target="_blank">W-018262</a>

**Comment by davidhabib** [comment]
am I correct that the report is not working just from the Campaign?  when I test invoking the report from a specific Job or Shift, it does seem to provide the correct filtered results.

**Comment by judisohn** [comment]
Right, because you have a specified record filter on a parent record (Shift or Job). The filter on the child object is basically ignored. But if the only thing you're filtering on is the record that is WITH or WITHOUT, the filter will fail. 

Try creating a Custom Report Type that's 3 levels and have the final join be 'with at least one record from (B) object' Create a report and filter on any value that's in C object. It will return just those results. Leave that tab open. Then go to the CRT and change the final join to 'with or without related records from (B) object.' Save and return to report you just ran, change nothing, and click Run. You'll get all (B) object records returned that are related to (A).

**Comment by coriobriensfdo** [comment]
This is working in Classic, not Lightning, and I believe is related to this similar issue for NPSP: https://github.com/SalesforceFoundation/Cumulus/issues/4018
